Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/test/core/TestCertificateRepository.kt

```diff
@@ -98,7 +100,13 @@ class TestCertificateRepository @Inject constructor(
         Timber.tag(TAG).d("requestCertificate(test.identifier=%s)", test.identifier)
 
         val newData = internalData.updateBlocking {
-            if (values.any { it.registrationToken == test.registrationToken }) {
+
+            val matchesExisting = values
+                .map { it.data }
+                .filterIsInstance<RetrievedTestCertificate>()
+                .any { it.registrationToken == test.registrationToken }
+
+            if (matchesExisting) {
                 Timber.tag(TAG).e("Certificate entry already exists for %s", test.identifier)
                 throw IllegalArgumentException("A certificate was already created for this ${test.identifier}")
```

Comment:
InvalidTestCertificateException should be used here

Reply:
I tried registering the same test certificate twice and got 'Collection contains more than one matching element'. Not sure where it is thrown